src/haven/instrument/robot.py

```diff
@@ -0,0 +1,124 @@
+import asyncio
+import logging
+
+from ophyd import Component as Cpt
+from ophyd import DynamicDeviceComponent as DCpt
+from ophyd import Device, EpicsMotor, EpicsSignal
+
+from .._iconfig import load_config
+from .device import aload_devices, make_device
+
+log = logging.getLogger(__name__)
+
+def transfer_sample(num_dios: int):
+    """Create a dictionary with robot sample device definitions.
+    For use with an ophyd DynamicDeviceComponent.
+    Parameters
+    ==========
+    num_dios
+      How many samples to create.
+    """
+    sample = {}
+    sample_attrs = ['present', 'empty', 'load', 'unload','x', 'y', 'z', 'rx', 'ry', 'rz']
+    for n in range(num_dios):
+        for attr in sample_attrs:
+            sample[f"sample{n}_{attr}"] = Cpt(EpicsSignal, f":sample{n}:{attr}", labels={"transfer"})
+    return sample
+
+
+class Robot(Device):
+    # joints and position
+    i = Cpt(EpicsMotor, ":i", labels={ "joint"})
+    j = Cpt(EpicsMotor, ":j", labels={ "joint"})
+    k = Cpt(EpicsMotor, ":k", labels={ "joint"})
+    l = Cpt(EpicsMotor, ":l", labels={ "joint"})
+    m = Cpt(EpicsMotor, ":m", labels={ "joint"})
+    n = Cpt(EpicsMotor, ":n", labels={ "joint"})
+    x = Cpt(EpicsMotor, ":x", labels={ "joint"})
+    y = Cpt(EpicsMotor, ":y", labels={ "joint"})
+    z = Cpt(EpicsMotor, ":z", labels={ "joint"})
+    rx = Cpt(EpicsMotor, ":rx", labels={ "joint"})
+    ry = Cpt(EpicsMotor, ":ry", labels={ "joint"})
+    rz = Cpt(EpicsMotor, ":rz", labels={ "joint"})
+    acc = Cpt(EpicsMotor, ":acceleration", labels={ "joint"})
```

Comment:
Acc and vel should probably be kind="config", right? They're unlikely to change during a scan.